src/main/java/com/softserve/itacademy/service/implamentations/GroupServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.softserve.itacademy.service.implamentations;
+
+import com.softserve.itacademy.entity.Group;
+import com.softserve.itacademy.dto.GroupDto;
+import com.softserve.itacademy.exception.NotFoundException;
+import com.softserve.itacademy.repository.GroupRepository;
+import com.softserve.itacademy.service.GroupService;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class GroupServiceImpl implements GroupService {
+
+    private final GroupRepository groupRepository;
+
+    public GroupServiceImpl(GroupRepository groupRepository) {
+        this.groupRepository = groupRepository;
+    }
+
+    @Override
+    public List<GroupDto> findAll() {
+        return groupRepository.findAll().stream().map(GroupDto::create).collect(Collectors.toList());
+    }
+
+    @Override
+    public void delete(Integer id) {
+        groupRepository.delete(getById(id));
+    }
+
+    @Override
+    public void updateDisabled(Integer id, Boolean disabled) {
+        Group group = getById(id);
+        group.setDisabled(disabled);
+        groupRepository.save(group);
```

Comment:
Wrong approach. You don't have to select and then save the whole object updating only one attribute. You are wasting the resources Instead just add appropriate method to the repo with the custom query (Update table set disabled=:some_param) and use this method whenever you need